pkg/cloudprovider/aws/instance.go

```diff
@@ -35,19 +40,69 @@ const (
 )
 
 type InstanceProvider struct {
-	ec2api ec2iface.EC2API
+	ec2api               ec2iface.EC2API
+	instanceTypeProvider *InstanceTypeProvider
 }
 
 // Create an instance given the constraints.
-// instanceTypeOptions should be sorted by priority for spot capacity type.
-// If spot is not used, the instanceTypeOptions are not required to be sorted
+// instanceTypes should be sorted by priority for spot capacity type.
+// If spot is not used, the instanceTypes are not required to be sorted
 // because we are using ec2 fleet's lowest-price OD allocation strategy
 func (p *InstanceProvider) Create(ctx context.Context,
 	launchTemplate *LaunchTemplate,
-	instanceTypeOptions []cloudprovider.InstanceType,
+	instanceTypes []cloudprovider.InstanceType,
 	subnets []*ec2.Subnet,
 	capacityType string,
-) (*string, error) {
+) (*v1.Node, error) {
+	// 1. Launch Instance
+	id, err := p.launchInstance(ctx, launchTemplate, instanceTypes, subnets, capacityType)
+	if err != nil {
+		return nil, err
+	}
+	// 2. Get Instance with backoff retry since EC2 is eventually consistent
+	instance := &ec2.Instance{}
+	if err := retry.Do(
+		func() (err error) { return p.getInstance(ctx, id, instance) },
+		retry.Delay(1*time.Second),
+		retry.Attempts(3),
```

Comment:
Do you have a reason why you picked 3 attempts of 1 second each? It seems pretty likely for EC2 to throttle for 3 seconds if you're making a lot of requests. 

Can we make this a larger interval with more retries? If an instance creation fails, we'll have to do binpacking all over again anyways and attempt to create the instance.

Reply:
I haven't seen any throttling failures here, even with some preliminary 1k node scale tests. I've never seen this one fail due to rate limiting, as the describe limits are actually quite high. I'm open to increasing these, but I'm a bit wary of the retries themselves causing cascading effects on other threads. Regardless, we will recover gracefully w/ the reallocator if these retries do fail.

Reply:
Cool. We should definitely keep track of this somewhere, especially when we get to larger cluster sizes.